service/pins.js

```diff
@@ -229,6 +229,22 @@ const deriveStatus = ({ name = "" }) => {
   return STATUS.queued
 }
 
+/**
+ * Parse the input using Date.parse, with special casing if the input is already a Date.
+ * 
+ * @param {Date|string} d - a Date object or ISO-formatted date string.
+ * @returns {number} - the number of milliseconds elapsed since January 1, 1970 00:00:00 UTC
+ */
+const parseDate = (d) => {
+  if (d instanceof Date) {
+    // Calling Date.parse on a Date instance causes the milliseconds to be thrown away,
+    // so e.g. a date representing 1616522896307 becomes 1616522896000.
+    // Calling toISOString and parsing _that_ seems to work. 
+    return Date.parse(d.toISOString())
```

Comment:
Could not this just do ?

```suggestion
    return d
```

Reply:
Huh, I guess so. I did the awkward string cast because `Date.parse` returns a number instead of a `Date` object & it seemed more consistent... but apparently you can compare `Date` and numbers using `<`, etc and it does the right thing.